custom/icds/location_rationalization/views.py

```diff
@@ -1,120 +1,115 @@
+        return reverse(LocationsListView.urlname, args=[self.domain])
 
     @property
     def page_context(self):
         context = super().page_context
-        context['form'] = self.form
+        context.update({
+            'bulk_upload': {
+                "download_url": reverse('download_location_rationalization', args=[self.domain]),
+                "adjective": _("locations"),
+                "plural_noun": _("location operations"),
+                "help_link": "TODO",
+            },
+        })
+        context.update({
+            'bulk_upload_form': get_bulk_upload_form(context),
+            'loc_types': self._location_types,
+        })
         return context
 
-    @property
-    @memoized
-    def form(self):
-        if self.request.POST:
-            return LocationRationalizationValidateForm(self.request.POST, self.request.FILES,
-                                                      location_types=self._location_types)
-        return LocationRationalizationValidateForm()
-
     def post(self, request, *args, **kwargs):
-        if self.form.is_valid():
-            transitions, errors = self._parse_upload()
-            [messages.error(request, error) for error in errors]
-            if not errors:
-                return self._generate_response(transitions)
+        try:
+            workbook = get_workbook(request.FILES['bulk_upload_file'])
+        except WorkbookJSONError as e:
+            messages.error(request, str(e))
+        else:
+            if self._workbook_is_valid(request, workbook):
+                transitions, errors = Parser(workbook.worksheets[0], self._location_types).parse()
+                [messages.error(request, error) for error in errors]
+                if not errors:
+                    return self._generate_response(transitions)
         return self.get(request, *args, **kwargs)
 
-    def _parse_upload(self):
-        uploaded_file = self.form.cleaned_data.get('file')
-        worksheet = get_workbook(uploaded_file).worksheets[0]
-        return Parser(worksheet, self._location_types).parse()
+    def _workbook_is_valid(self, request, workbook):
+        # ensure mandatory columns in the excel sheet
+        worksheet = workbook.worksheets[0]
+        headers = worksheet.fieldnames
+        expected_headers = []
+        for location_type in self._location_types:
+            expected_headers.extend([f'old_{location_type}', f'new_{location_type}'])
+        missing_headers = set(expected_headers) - set(headers)
+        if missing_headers:
+            messages.error(request, _("Missing following columns in sheet: {columns}").format(
+                columns=", ".join(missing_headers)
+            ))
+            return False
+        return True
 
     @cached_property
     def _location_types(self):
-        location_types = [lt.code for lt in LocationType.objects.by_domain(self.domain)]
+        location_types = LocationType.objects.by_domain(self.domain)
```

Comment:
this does not look right. it should be the codes only.

Reply:
Thank, yep, not intended. 6893fe6b31a

Reply:
This breaks the location type dropdown, which needs the type name. Any objection to me changing it back?

Reply:
Added 06525be14b5 to https://github.com/dimagi/commcare-hq/pull/26865, though I'd rather show the type names than just the codes.

Reply:
Oh i see, ya i don't mind changing it back, just would need to change where its called as well.